File: SobreCarga.py
import math
class FiguraGeometrica:
    def area_circulo(self, radio):
        return math.pi * radio * radio  # circulo

    def area_rectangulo(self, base, altura):
        return base * altura  # rectangulo

    def area_triangulo(self, base, altura):
        return (base * altura) / 2  # triangulo rectangulo

    def area_trapecio(self, base_mayor, base_menor, altura):
        return ((base_mayor + base_menor) * altura) / 2  # trapecio

    def area_pentagono(self, lado, apotema):
        return (5 * lado * apotema) / 2  # pentagono

f = FiguraGeometrica()

print("circulo:", f.area_circulo(1))
print("rectangulo:", f.area_rectangulo(4, 3))
print("triangulo:", f.area_triangulo(4, 8))
print("trapecio:", f.area_trapecio(2, 4, 6))
print("pentagono:", f.area_pentagono(5, 3))

File: SobreCarga.java
package inf121;

public class FiguraGeometrica {
	
	double area(double radio){
		return Math.PI * radio * radio;//circulo
	}
	
	double area(float b, double a) {
		return b * a; //rectangulo
	}
	
	double area(double b, double a) {
		return (b * a)/2; //triangulo rectangulo
	}
	
	double area(double base, float basemenor, double altura) {
		return ((base + basemenor) * altura)/2; //trapecio
	}
	
	 double area(float lado, float apotema) {
	        return (5 * lado * apotema)/2; //pentagono
	    }
	
	public static void main(String[] args) {
		FiguraGeometrica f1 = new FiguraGeometrica();
		FiguraGeometrica f2 = new FiguraGeometrica();
		FiguraGeometrica f3 = new FiguraGeometrica();
		FiguraGeometrica f4 = new FiguraGeometrica();
		FiguraGeometrica f5 = new FiguraGeometrica();
		System.out.println("circulo: " + f1.area(1));
		System.out.println("rectangulo: " + f2.area(4,3));
		System.out.println("triangulo: " + f3.area(4,8));
		System.out.println("trapecio: " + f4.area(2,4,6));
		System.out.println("pentagono: " + f5.area(5,3));
	}
}
